### ISSUE TITLE
Deleting a single type after renaming it caused an unrecoverable error.

### ISSUE THREAD

**Issue by FieldMarshallVague** [strapi/strapi#14131]
- Node.js version:  16.14.2
- NPM version: 8.7.0
- Strapi version: 4.3.2
- Database: postgres
- Operating system: win 10 pro 64

### Describe the bug

I renamed a single type from customer-story to customer-stories.  The server restarted and everything seemed normal.  I tried to retrieve that type and realised the API endpoint was still the old name, so I looked at the content editor and it seemed I couldn't change it.  So I deleted it using the edit > delete link.  The server showed the busy state for a while, then said it was taking too long and I should check the logs.  The server had stopped with error code 1.

I tried restarting and got the same error.  I deleted cache and dist folders and rebuilt, but same error.  Eventually, I rolled back the changes in git to before the rename and rebuilt and that worked to get it building and running, but I still can't delete that single type.

### Steps to reproduce

Create CustomerStory type as a single type.
Create some fields and populate with dummy data.
Save/restart.
Change name of type to 'CustomerStories'
Try to fetch data for CustomerStories via api.
(This should work, but using 'CustomerStory' endpoint, not CustomerStories.)
Delete CustomerStories type.
This should error.

### Expected behavior

I would expect to be able to rename the endpoint when renaming the single type, but otherwise that it would be delete-able without crashing, so I can start over.

### Additional context

I suspect the problem has to do with my rename clashing with the collection name (I hadn't created a collection type, but it does it automatically, at least as far as the schema.json is concerned).  Even though I have rolled back, my schema.jsons shows the collection type for customer-story as 'customer-stories'.  So I guess the singular and collection both had the same name when I went to do the delete.

**Comment by bolg55** [comment]
Can you amend this to include explicit steps to reproduce? Thanks

**Comment by FieldMarshallVague** [comment]
Done. Sorry, I thought it was clear enough but the above should help.

**Comment by petersg83** [comment]
Hello,
Is there an error message?

**Comment by FieldMarshallVague** [comment]
Yes, the one above.  "The server had stopped with error code 1."

**Comment by jhoward1994** [comment]
Hi FieldMarshallVague,

Thanks for your patience.

- It is expected that renaming the single type from `CustomerStory` to `CustomerStories` does not change the API endpoint. For single types the endpoints are based on the `singularName` whereas it sounds like you had changed the `displayName`
- I'm afraid I was unable to replicate the crash you reported, deleting the singleType worked fine for me before and after renaming. If this is a persistent issue for you could you please provide more specific steps in order to replicate the crash?

**Comment by FieldMarshallVague** [comment]
hi @jhoward1994 ,  I changed the name via the Edit button in the single type's content editor.  I've just checked and this is the display name. 

Just to be clear, I didn't have a problem with the api endpoint, that was just a test that it was working (sorry, I see this might be confusing).

I didn't mention, but it occurs to me now that it might be relevant:  I upgraded from v3 using the CodeMods tool and following the migrations guides.  I believe it was 4.1.0 when I started.  This may not be an issue with newer migrations.  

I'm happy for you to close this and I'll re-open if I get any further info.
